test/core/tsi/ssl_transport_security_test.cc

```diff
@@ -308,10 +309,12 @@ static void ssl_test_check_handshaker_peers(tsi_test_fixture* fixture) {
     check_session_reusage(ssl_fixture, &peer);
     check_alpn(ssl_fixture, &peer);
     check_security_level(&peer);
-    if (ssl_fixture->server_name_indication != nullptr) {
-      check_server1_peer(&peer);
-    } else {
+    if (ssl_fixture->server_name_indication == nullptr ||
+        strcmp(ssl_fixture->server_name_indication, SSL_TSI_TEST_WRONG_SNI) ==
+            0) {
```

Comment:
I am a little bit confused about the logic here. When we see the fixture is setting a SNI, we call `check_server0_peer `, but in `check_server0_peer ` I don't see it check against the SNI-specific information.  Do I misunderstand something here?

Reply:
The test has the following setup: server has two creds, server0 and server1. When providing correct SNI on server1, it will check peer against server1 credential. If SNI is nullptr or a wrong SNI, then we expect the default credentials (server0) will be picked, thus check peer against server0 credential.

Reply:
Got it, thanks!